core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -358,14 +358,12 @@ private[spark] class MesosClusterScheduler(
         .orElse(desc.command.environment.get("SPARK_EXECUTOR_URI"))
   }
 
-  private def getDriverEnvironment(desc: MesosDriverDescription): Environment = {
-    val env = {
-      val executorOpts = desc.conf.getAll.map { case (k, v) => s"-D$k=$v" }.mkString(" ")
-      val executorEnv = Map("SPARK_EXECUTOR_OPTS" -> executorOpts)
```

Comment:
Not only are these variables already set in `generateCmdOption`, but SPARK_EXECUTOR_OPTS has no effect on the driver process